serde_v8/magic/resource.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::magic::transl8::magic_deserialize;
+use crate::magic::transl8::magic_serialize;
+use crate::magic::transl8::FromV8;
+use crate::magic::transl8::MagicType;
+use crate::magic::transl8::ToV8;
+use std::cell::Cell;
+use std::ffi::c_void;
+use std::marker::PhantomData;
+use std::ptr::NonNull;
+use std::rc::Rc;
+
+/// A Resource is a Rust object managed by the V8 GC.
+/// `T` is reference counted using an `Rc<T>`.
+/// When sent to V8, the Resource will be wrapped in a JavaScript object.
+/// The JavaScript object will hold a reference to the Rust object.
+///
+/// The underlying Rc<T> will always have a strong count >= 1 until either
+/// the JavaScript object is garbage collected OR `into_inner` is called.
+pub struct Resource<T: ?Sized> {
```

Comment:
Can the whole thing be simplified by using `inner: Rc<RefCell<Option<T>>>` maybe?